### PR TITLE
[HGCal trigger] Remove deprecated v8 geometry code + Add new trigger geometry class

### PR DESCRIPTION
**Pull request by jbsauvan**
#### PR description:
- Remove deprecated code used for the HGCAL V8 geometry (related to issue #27095 )
- Add a new trigger geometry class `HGCalTriggerGeometryV9Imp3` containing information on the connections between modules, lpGBT links and backend FPGAs, read from a json file. 
Currently not used by default, but will replace `HGCalTriggerGeometryV9Imp2` in the future. 

Associated internal PRs and reviews:
- PFCal-dev#466
- PFCal-dev#471
- PFCal-dev#482
- PFCal-dev#491
- PFCal-dev#493

This PR depends on external cms-data/L1Trigger-L1THGCal#22

#### PR validation:
Private tests in `L1Trigger/L1THGCalUtilities/test`:
```
cmsRun testHGCalL1T_RelValV11_cfg.py
cmsRun testHGCalL1T_multialgo_V11_cfg.py
```
Tested D49, D60 and D68 workflows
```
runTheMatrix.py -w upgrade -l 23234.0 --maxSteps=2 
runTheMatrix.py -w upgrade -l 28234.0 --maxSteps=2 
runTheMatrix.py -w upgrade -l 31434.0 --maxSteps=2
```
---
@snwebb FYI

### CODE REVIEW

**Comment by perrotta** [comment]
test parameters:
- pull_request = cms-data/L1Trigger-L1THGCal#22

**Comment by cecilecaillol** [comment]
please test with https://github.com/cms-data/L1Trigger-L1THGCal/pull/22

**Comment by civanch** [comment]
+1

**Comment by cecilecaillol** [comment]
+l1

File: L1Trigger/L1THGCal/src/HGCalTriggerTools.cc [hunk 1]
```diff
@@ -42,17 +40,12 @@ void HGCalTriggerTools::eventSetup(const edm::EventSetup& es) {
...
 
-  if (geom_->isV9Geometry()) {
-    bhLayers_ = geom_->hscTopology().dddConstants().layers(true);
-    totalLayers_ = eeLayers_ + fhLayers_;
-  } else {
-    bhLayers_ = geom_->bhTopology().dddConstants()->getMaxDepth(1);
-    totalLayers_ = eeLayers_ + fhLayers_ + bhLayers_;
-  }
+  bhLayers_ = geom_->hscTopology().dddConstants().layers(true);
+  totalLayers_ = eeLayers_ + fhLayers_;
```
Comment (srimanob) [inline L1Trigger/L1THGCal/src/HGCalTriggerTools.cc:44]: Is the definition of totalLayers_ changed here?

**Comment by srimanob** [comment]
Just wonder, in addition of the PR, do you still want to keep `FIXME` of HFNose in the following line,
https://github.com/cms-sw/cmssw/blob/6c52b5ba1a941474fd593115b567e2956b0c5800/L1Trigger/L1THGCal/src/HGCalTriggerTools.cc#L94

File: L1Trigger/L1THGCal/src/HGCalTriggerTools.cc (hunk 1 above)
Comment (jbsauvan) [reply L1Trigger/L1THGCal/src/HGCalTriggerTools.cc:44]: The definition of `totalLayers_` as `totalLayers_ = eeLayers_ + fhLayers_ + bhLayers_` was only for geometries <=V8, where the HCAL layers were divided into front and back sections. In geometries >= V9, the silicon section of HCAL is covering all the HCAL layers (given by `fhLayers_`). 
Before there was a switch to check whether the geometry was <=V8 or >=V9; now the switch has been removed. So the definition of `totalLayers_` for geometries >=V9 has not changed.

**Comment by jbsauvan** [comment]
> Just wonder, in addition of the PR, do you still want to keep `FIXME` of HFNose in the following line,
> 
> https://github.com/cms-sw/cmssw/blob/6c52b5ba1a941474fd593115b567e2956b0c5800/L1Trigger/L1THGCal/src/HGCalTriggerTools.cc#L94

The method `HGCalTriggerTools::layers(ForwardSubdetector type)` is actually used for `HFNose`. The function using `DetId::Detector type` can't easily handle `HFNose` I think, I'll have to think about what to do there.

**Comment by srimanob** [comment]
+Upgrade

**Comment by qliphy** [comment]
+1

**Pull request merged**